components/raftstore/src/store/worker/split_controller.rs

```diff
@@ -247,15 +247,16 @@ impl Recorder {
             |x| x,
         );
         let mut samples = Samples::from(sampled_key_ranges);
+        let recorded_key_ranges: Vec<&KeyRange> = self.key_ranges.iter().flatten().collect();
         // Because we need to observe the number of `no_enough_key` of all the actual keys,
         // so we do this check after the samples are calculated.
-        if (self.key_ranges.len() as u64) < config.sample_threshold {
+        if (recorded_key_ranges.len() as u64) < config.sample_threshold {
```

Comment:
How about using `size_hint`?

Reply:
`size_hint`'s output is a tuple and has an `Option` inside it, I think using `collect` is more simple here.

Reply:
I think maybe you can use `.0` so that we can avoid `collect` and `into_iter`.

Reply:
Also, after `flatten()`, the result of `size_hint()` will be `(0, None)`.

![origin_img_v2_85e192ec-377a-449b-bb6e-51ce6b1730ag](https://user-images.githubusercontent.com/1446531/155324073-c7ed14fe-0974-4129-b244-fa59a74d423c.png)